.github/docker_images/abidiff/diff.sh

```diff
@@ -0,0 +1,27 @@
+#!/usr/bin/env bash
+
+# Copyright Amazon.com, Inc. or its affiliates. All Rights Reserved.
+# SPDX-License-Identifier: Apache-2.0 OR ISC
+
+set -ex
+
+cd /previous && /build.sh
+cd /next && /build.sh
+
+CHECK_LIB="$1"
+ARTIFACT_PATH="build"
+if [[ "${CHECK_LIB}" == "crypto" ]]; then
+  ARTIFACT_PATH="${ARTIFACT_PATH}/crypto/libcrypto.so"
+elif [[ "${CHECK_LIB}" == "ssl" ]]; then
+  ARTIFACT_PATH="${ARTIFACT_PATH}/ssl/libssl.so"
+else
+  exit 1
+fi
+
+set +e
+abidiff --hd1 "/previous/include" --hd2 "/next/include" "/previous/${ARTIFACT_PATH}" "/next/${ARTIFACT_PATH}"
```

Comment:
Does the fact that "crypto" and "ssl" headers are mixed potentially affect the results when the test should only be considering one of them?

Reply:
No, its just a means to do an initial pass at filtering the symbols in the library, and only focusing on the ones that match with those defined in the headers. So think something along the lines of `nm <library so> | grep <symbols from headers>`